api/resources_portal/views/document_views.py

```diff
@@ -0,0 +1,330 @@
+        return result
+
+
+class OrganizationDocumentSerializer(serializers.Serializer):
+    id = serializers.IntegerField(read_only=True)
+    name = serializers.CharField(read_only=True)
+    owner_id = serializers.CharField(read_only=True)
+    created_at = serializers.DateField(read_only=True)
+    updated_at = serializers.DateField(read_only=True)
+
+    class Meta:
+        model = Organization
+        fields = (
+            "id",
+            "name",
+            "owner_id",
+            "created_at",
+            "updated_at",
+        )
+        read_only_fields = ("id", "created_at", "updated_at")
+
+
+class OrganizationDocumentView(DocumentViewSet):
+    """Defines search and filter fields for Organizations"""
+
+    document = OrganizationDocument
+    serializer_class = OrganizationDocumentSerializer
+    pagination_class = ESLimitOffsetPagination
+
+    # Filter backends provide different functionality we want
+    filter_backends = [
+        FilteringFilterBackend,
+        OrderingFilterBackend,
+        DefaultOrderingFilterBackend,
+        CompoundSearchFilterBackend,
+    ]
+
+    lookup_field = "id"
+
+    # Define search fields
+    # Is this exhaustive enough?
+    search_fields = {"name": {"boost": 10}}
+
+    # Define filtering fields
+    filter_fields = {}
+
+    # Define ordering fields
+    ordering_fields = {
+        "name": "name",
+        "id": "id",
+    }
+
+    # Specify default ordering
+    ordering = ("_score", "created_at")
+
+    faceted_search_fields = {}
+
+    faceted_search_param = "facet"
+
+
+class UserDocumentSerializer(serializers.Serializer):
+    id = serializers.IntegerField(read_only=True)
+    username = serializers.CharField(read_only=True)
+    first_name = serializers.CharField(read_only=True)
+    last_name = serializers.DateField(read_only=True)
+    created_at = serializers.DateField(read_only=True)
+    updated_at = serializers.DateField(read_only=True)
+
+    class Meta:
+        model = User
+        fields = (
+            "id",
+            "username",
+            "first_name",
+            "last_name",
+            "created_at",
+            "updated_at",
+        )
+        read_only_fields = ("username", "created_at", "updated_at")
+
+
+class UserDocumentView(DocumentViewSet):
+    """Defines search and filter fields for Users"""
+
+    document = UserDocument
+    serializer_class = UserDocumentSerializer
+    pagination_class = ESLimitOffsetPagination
+
+    # Filter backends provide different functionality we want
+    filter_backends = [
+        FilteringFilterBackend,
+        OrderingFilterBackend,
+        DefaultOrderingFilterBackend,
+        CompoundSearchFilterBackend,
+    ]
+    ordering = ("created_at",)
+    lookup_field = "id"
+
+    # Define search fields
+    # Is this exhaustive enough?
+    search_fields = {
+        "username": {"boost": 10},
+        "last_name": {"boost": 8},
+        "first_name": {"boost": 7},
+    }
```

Comment:
@dvenprasad will we want to support searching by email addresses?

Reply:
Not for materials. In the future, we would want to be able to search for people by email, for adding them to organizations or looking people up to join their organization.

Reply:
I added email to searchable fields here just in case.